### PR TITLE
lint: Change go linter from gometalinter to golangci-lint

### PR DESCRIPTION
**Pull request by ganeshmaharaj**
gometalinter is being deprecated and will be archived April '19. Their suggestion is to switch to using golangci-lint.

Signed-off-by: Ganesh Maharaj Mahalingam <ganesh.mahalingam@intel.com>

### CODE REVIEW

**Comment by ganeshmaharaj** [comment]
I tried running the runtime code through the new tool and this is as much progress i have made. There is still loads of places that needs to be cleaned up. I am no expert in golang and I thought this PR will help me course correct if i am heading in the wrong direction with some of my choices. the following feedback would really help.

1. Am I heading in the right direction?
2. Are there choices on how we handle errors in some of the defer functions where there is no logger available.
3. Any other feedback?

**Comment by jodh-intel** [comment]
@ganeshmaharaj - Travis is barfing on:

```
ERROR: Commit 7d25e219cb646e0538ed94f476f14eeb72957fab: Failed to find subsystem in subject: "Change go linter from gometalinter to golangci-lint"
ERROR: checkcommits failed. See the document below for help on formatting
commits for the project.
https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#patch-format
```

Please could you create an issue for this PR then update your commit message to something like:

```
lint: Update code for new golangci-lint linter

gometalinter is now deprecated so update the code for the new `golangci-lint linter`.

Fixes #XXX.

Signed-off-by: Ganesh Maharaj Mahalingam <ganesh.mahalingam@intel.com>
```

**Comment by ganeshmaharaj** [comment]
@jodh-intel +1. thanks for the review. i will leave a `.golangci.yml` file in each repository and when we switch to the new tool, it will use a per-repository config to run tests on each of them. Upside, each repository is self contained. Downside, if we want to change something in the code, we need to fix it in all the repositories.

**Comment by jodh-intel** [comment]
@ganeshmaharaj - I still haven't looked at this new linter I'm afraid, but I think we should be able to maintain a single `.golangci.yml` file in the `tests` repo (which would be the ideal).

Each repo has a `.ci/` directory which contains scripts that:

- clone the `tests` repo.
- runs the `.ci/run.sh` script in the `tests` repo.

That script then "calls back" into the original repo by invoking an appropriate `make` call.

It's a bit *sleight of hand* but it does allow us to centralise 99.99% of our CI and (non-unit test) test material.

**Comment by ganeshmaharaj** [comment]
The new linter has been switched in tests. i think without this change, all new PRs would fail for the runtime repository. Kindly request you to expedite this change.

**Comment by teawater** [comment]
```
/home/travis/gopath/src/github.com/kata-containers/runtime/virtcontainers/utils
cli/kata-check_ppc64le_test.go:125:16: struct of size 40 bytes could be of size 32 bytes (maligned)
	type testData struct {
	              ^
virtcontainers/hypervisor_test.go:448:6: `genericTestRunningOnVMM` is unused (deadcode)
func genericTestRunningOnVMM(t *testing.T, data []testNestedVMMData) {
     ^
cli/kata-check.go:246:6: `genericHostIsVMContainerCapable` is unused (deadcode)
func genericHostIsVMContainerCapable(details vmContainerCapableDetails) error {
     ^
The command ".ci/static-checks.sh" failed and exited with 1 during .
Your build has been stopped.
```

**Comment by ganeshmaharaj** [comment]
@teawater can you please tell me how to reproduce your test results. Neither the CI nor are my test producing the same results.

**Comment by chavafg** [comment]
I think this comes from the PowerPC job on travis:
https://travis-ci.org/kata-containers/runtime/jobs/510218072

File: .travis.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ os:
 go_import_path: github.com/kata-containers/runtime
 
 go:
-  - "1.10.x"
+  - "1.11.x"
```
Comment (ganeshmaharaj) [inline .travis.yml]: @gnawux @teawater is this change something you guys agree to or would you prefer i move this out to a seperate patch where we update the baseline for all kata  code-bases?

**Comment by ganeshmaharaj** [comment]
This seems to be the issue with the failing test. 
```
level=info msg="[loader] Go packages loading at mode load deps types and syntax took 855.769539ms"
level=error msg="Running error: context loading failed: failed to load program with go/packages: go [list -e -json -compiled -test=true -export=false -deps=true -find=false -- ./...]: exit status 1: go: disabling cache (/home/jenkins/.cache/go-build) due to initialization failure: open /home/jenkins/.cache/go-build/log.txt: permission denied\ngo list -compiled requires build cache\n"
level=info msg="Memory: 11 samples, avg is 69.7MB, max is 69.7MB"
level=info msg="Execution took 1.034841214s"
Makefile:2: recipe for target 'test' failed
make: *** [test] Error 3
Build step 'Execute shell' marked build as failure
```
Is this a known issue? Should we merge the patch to allow other changes to merge?

**Comment by teawater** [comment]
```
+ res=1
+ '[' 1 -eq 0 ']'
+ die 'template factory is not cleaned up'
+ msg='template factory is not cleaned up'
+ echo 'ERROR: template factory is not cleaned up'
ERROR: template factory is not cleaned up
+ exit 1
Makefile:123: recipe for target 'vm-factory' failed
make: *** [vm-factory] Error 1
Build step 'Execute shell' marked build as failure
Performing Post build task...
Match found for :.* : True
Logical operation result is TRUE
Running script  : #!/bin/bash
```
I reproduced this issue in my part a lot of times.  It is a random failure of vm-factory.
So I think this PR does not relate to this fail.

**Pull request merged**